o3de/o3de
# new: Allow to use the LegacyComponentConverter between a lumberyard and o3de project

## What is this tool ?

The LegacyComponentConverter convert lumberyard material/mesh components to Atom components for given levels and slices. 

It is designed to be run against a lumberyard project and will do the conversion inline. By itself it is not useable as Atom is not in the lumberyard repo and o3de cannot read .ly and .slice files. You need to use the [SliceConverter](https://github.com/o3de/community/tree/main/Migration%20Guides/Lumberyard%20to%20O3DE/Misc-Documents/Slice-Converter) afterwards to create .prefab files that can be imported to o3de.

Outdated documentation [here](https://github.com/o3de/community/tree/main/Migration%20Guides/Lumberyard%20to%20O3DE/Misc-Documents/Tutorials/Starter-Game-On-Atom) (to be updated in another PR)

## What does this PR do?

In a nutshell, it allows to import any level from lumberyard to o3de with correct transform, meshes and materials still assigned to entities.

This is driven by the addition of the `assetCatalogOverridePath` argument. This args allows you to use a given o3de project catalog as a conversion target. It will match entities based on relative asset path, and use your o3de project asset ids during the conversion.

(your o3de project needs to have imported the .fbx and .material assets via the tool fixed in https://github.com/o3de/o3de/pull/17377)

## How was this PR tested?

1. As stated in the tool readme, needs a custom build of o3de and lumberyard to save assetCatalog as xml instead of binary

2. Created a simple level in lumberyard with .fbx files and a slice. Closed lumberyard and asset processor.

![lumberyard](https://github.com/o3de/o3de/assets/19243508/6cc10481-d740-489f-a983-6dfd85c1eda8)

3. Ran the asset conversion tool, placed the asset in a new o3de project and opened it once to populate the assetCatalog.

4. Run this tool (launch file sample below)

![image](https://github.com/o3de/o3de/assets/19243508/732b7f44-290b-459c-ba89-ee9cecbfdc69)

5. Run the Slice converter (launch args sample below). Will trigger a few assertion but you can continue, it does work

![image](https://github.com/o3de/o3de/assets/19243508/94188590-cb00-477f-8523-511b5ebf40f8)

6. Copy the generated .prefab files to o3de and open the level. Need to import a basic scene prefab else no lighting.

![o3de](https://github.com/o3de/o3de/assets/19243508/24e08e88-0f90-4a1a-a16b-09cef820188a)

In the image above, entities where moved to the center of the scene, nothing else was modified (no component change, no asset/entity creation or tinkering)

Further testing incomming with larger scene, the goal being to see how much of the StarterGame level can be ported automatically
